logger.go

```diff
@@ -22,13 +22,16 @@ type Logger struct {
 	labels Labels
 }
 
-// WithLabels returns a logger whose module is the same
-// as this logger and the returned logger will add the
-// specified labels to each log entry.
+// WithLabels returns a logger whose module is the same as this logger and
+// the returned logger will add the specified labels to each log entry.
+// WithLabels only target a specific logger with labels. Children of the logger
+// will not inherit the labels.
+// To add labels to all child loggers, use ChildWithLabels.
```

Comment:
Right, so this overwrites; children (below) inherit.

Reply:
Yes.

Reply:
We should clarify the doc string slightly.
When I read the above, it sounds like "will add the specified labels" implying that it will keep the labels that it has, but add the ones specified, but will not cause children of this to get the labels.
I think the actual practice of this is:

1) Fork the tree of modules, creating a new logger at the same module, but replacing its labels with the supplied labels
2) Children of the original logger will continue to inherit that logger's labels
3) Children of this logger will inherit only this loggers labels.

Is that an accurate statement? Is there a better way to describe this in the docstring?